packages/plugins/plugin-table/src/next/table.ts

```diff
@@ -0,0 +1,150 @@
+      const currentIndex = table.columnOrdering.indexOf(event.columnId);
+      if (currentIndex !== -1) {
+        table.columnOrdering.splice(currentIndex, 1);
+        table.columnOrdering.splice(event.newIndex, 0, event.columnId);
+      }
+      break;
+    }
+    case 'PinRow': {
+      table.pinnedRows[event.side].push(event.rowIndex);
+      break;
+    }
+    case 'UnpinRow': {
+      table.pinnedRows.top = table.pinnedRows.top.filter((index: number) => index !== event.rowIndex);
+      table.pinnedRows.bottom = table.pinnedRows.bottom.filter((index: number) => index !== event.rowIndex);
+      break;
+    }
+    case 'SelectRow': {
+      if (!table.rowSelection.includes(event.rowIndex)) {
+        table.rowSelection.push(event.rowIndex);
+      }
+      break;
+    }
+    case 'DeselectRow': {
+      table.rowSelection = table.rowSelection.filter((index: number) => index !== event.rowIndex);
+      break;
+    }
+    case 'DeselectAllRows': {
+      table.rowSelection = [];
+      break;
+    }
+    case 'ModifyColumnWidth': {
+      const columnId = table.columnOrdering.at(event.columnIndex);
+      if (columnId) {
+        const newWidth = Math.max(0, event.width);
+        table.columnWidths[columnId] = newWidth;
+      }
+      break;
+    }
+  }
+  return table;
+};
+
+// Return a preact-signals reactive object (with create) for an initialized table.
+// TODO(Zan): Take widths + callbacks for width changes, column ordering.
+export const createTable = (columnDefinitions: ColumnDefinition[], data: any[]) => {
+  /**
+   * Creates a computed signal structure for table cells.
+   *
+   * This structure is optimized for reactive collaboration on cell edits:
+   *
+   * 1. Outer computed:
+   *    - Tracks changes to the entire data array (additions, removals, reordering)
+   *    - Recreates all cell computed signals when triggered
+   *
+   * 2. Inner computed (one per cell):
+   *    - Tracks changes to individual cell data
+   *    - Recalculates only when its specific cell data changes
+   *
+   * Benefits:
+   * - Highly efficient for cell-level edits (most common in collaborative editing)
+   * - Maintains reactivity at the cell level
+   * - Automatically updates UI for all users on any collaborative change
+   * - Eliminates the need for a separate row structure
+   *
+   * Performance characteristics:
+   * - Cell edits: Optimal (only affected cell recalculates)
+   * - Array-level changes: More expensive (all cell `computed` recreated)
+   */
+
+  const cells = computed(() => {
+    const cellMap = new Map();
+
+    data.forEach((row, rowIndex) => {
+      columnDefinitions.forEach((col, colIndex) => {
+        const key = `${colIndex},${rowIndex}`;
+        cellMap.set(
+          key,
+          computed(() => col.accessor(row)),
+        );
+      });
+    });
+
+    return cellMap;
+  });
+
+  const rowCount = computed(() => data.length);
+  const updateTracker = new CellUpdateTracker(cells);
+
+  return create({
```

Comment:
why are you not using ECHO objects directly? the data should be the result of a query and not require mapping. similarly for other state (see Sheet)?